dtfs-central-api/src/helpers/get-fee-record-payment-entity-group-reconciliation-data.ts

```diff
@@ -0,0 +1,59 @@
+import { orderBy } from 'lodash';
+import { FEE_RECORD_STATUS } from '@ukef/dtfs2-common';
+import { TfmUsersRepo } from '../repositories/tfm-users-repo';
+import { FeeRecordReconciledByUser } from '../types/utilisation-reports';
+import { FeeRecordPaymentEntityGroup } from '../types/fee-record-payment-entity-group';
+
+/**
+ * Gets the reconciliation data for a fee record payment entity group
+ * @param group - The fee record payment entity group
+ * @returns The group reconciliation data
+ * @throws {Error} If the groups fee records array is empty
+ * @throws {Error} If any of the fee records at the `RECONCILED` status have a null `dateReconciled` property
+ */
+export const getFeeRecordPaymentEntityGroupReconciliationData = async (
+  group: FeeRecordPaymentEntityGroup,
+): Promise<{
+  reconciledByUser?: FeeRecordReconciledByUser;
+  dateReconciled?: Date;
+}> => {
+  if (group.feeRecords.length === 0) {
+    throw new Error('Fee record payment entity group cannot have an empty fee records array');
+  }
+
+  const reconciledFeeRecords = group.feeRecords.filter(({ status }) => status === FEE_RECORD_STATUS.RECONCILED);
+
+  if (reconciledFeeRecords.length === 0) {
+    return {};
+  }
+
+  if (reconciledFeeRecords.some(({ dateReconciled }) => dateReconciled === null)) {
+    throw new Error(`Fee records at the '${FEE_RECORD_STATUS.RECONCILED}' status cannot have a null 'dateReconciled' property`);
+  }
+
+  const feeRecordsSortedByDateReconciledDescending = orderBy(reconciledFeeRecords, [(feeRecord) => feeRecord.dateReconciled!.getTime()], ['desc']);
+  const mostRecentlyReconciledFeeRecord = feeRecordsSortedByDateReconciledDescending.at(0);
+  if (!mostRecentlyReconciledFeeRecord) {
+    // Based on all the above checks, this error should never be reached, hence not being documented
```

Comment:
can this line be removed please

Reply:
or instead a reason why this error should never be reached, instead "of not being documented"

Reply:
It shouldn't be reached "based on all the above checks", and restating how the above code feels redundant and verbose

Reply:
I don't think any documentation should say ".... which is why X is not documentated", it has no value.

Do we even need to throw an error here? Will the central API crash if this error occurs? It should continue to run.

Maybe it could conditionally execute instead like this?

```ts
if (mostRecentlyReconciledFeeRecord) {
  // await...
  // return {}
}
```

Reply:
Can we add some new lines after the consts on line 34 and 35 please